File: lib/utils/stats_utils.dart
class StatsUtils {
  const StatsUtils._();

  static double maxValue(Iterable<num> values, {double minimum = 0}) {
    var maxValue = minimum;
    for (final value in values) {
      if (value > maxValue) {
        maxValue = value.toDouble();
      }
    }
    return maxValue;
  }
}

File: app/src/main/java/io/github/benji377/timety/util/stats/StatsUtils.kt
package io.github.benji377.timety.util.stats


object StatsUtils {


    fun maxValue(values: Iterable<Number>, minimum: Double = 0.0): Double {
        var maxValue = minimum
        for (value in values) {
            val d = value.toDouble()
            if (d > maxValue) maxValue = d
        }
        return maxValue
    }
}
